feature-utils/silly-i18n/rollup-plugin.cjs

```diff
@@ -0,0 +1,57 @@
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+const glob = require("glob");
+
+const DIRNAME = "locales";
+const FILEEXT = ".json";
+const PLUGIN = "silly-i18n";
+const SUFFIX = `?${PLUGIN}`;
+
+exports.default = function (options) {
+    return {
+        name: PLUGIN,
+
+        async resolveId(source, importer) {
+            if (source == `!${PLUGIN}`) {
+                for (let dir = importer; dir != (dir = path.dirname(dir)); ) {
+                    const subdir = path.join(dir, DIRNAME);
+                    const exists = await fs.promises.stat(subdir).then(
+                        (stat) => stat.isDirectory(),
+                        () => false
+                    );
+                    if (exists) return `${subdir}${SUFFIX}`;
+                }
+                throw new Error(`"${DIRNAME}" directory not found`);
+            }
+        },
+
+        async load(id) {
+            if (id.endsWith(SUFFIX)) {
```

Comment:
Also, this is generating the combination of files *once* for every file, as far as I can tell; essentially generating an object multiple times. In cases like `fbImport` that can take a while, and it can *really* add to build time.

Reply:
I don't think it does, since the `id` (as returned by `resolveId`) is the absolute path to the specific `locales` directory (plus the `SUFFIX`). My understanding is that Rollup calls `load` only once for each unique `id`.